### PR TITLE
Issue #8664: NeedBraces throws a NPE when parsing switch expression syntax

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8664: NeedBraces throws a NPE when parsing switch expression syntax

Diff Regression projects: https://gist.githubusercontent.com/nmancus1/e3988f8092d919b5cbe70f5359c4d9e8/raw/a0f6f5860f8025c19868061dee6e0e40960b992f/projects-to-test-on.properties

Diff Regression config: https://gist.githubusercontent.com/nmancus1/9856763c94e3283016546a6dcd709208/raw/7e2b5b9c4f472f935ea48d958dbdd926e215aa2a/NeedBraces.xml

### CODE REVIEW

**Comment by nmancus1** [comment]
Github, generate report

**Comment by nmancus1** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 1]
```diff
@@ -476,26 +492,115 @@ private static DetailAST getLastLambdaToken(DetailAST lambda) {
...
+     * Checks if switch member (case or default statement) in a switch rule or
+     * case group is on a single line.
+     *
+     * @param statement {@link TokenTypes#LITERAL_CASE case statement} or
+     * {@link TokenTypes#LITERAL_DEFAULT default statement}.
+     * @return true if current switch member is single-line statement.
+     */
+    private static boolean isSingleLineSwitchMember(DetailAST statement) {
+        final boolean isInSwitchRule =
+            statement.getParent().getType() == TokenTypes.SWITCH_RULE;
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:548]: This code duplicates method `isSwitchRuleLambda`. Should this method be used here?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 2]
```diff
@@ -293,9 +298,19 @@ private boolean isEmptyLoopBodyAllowed(DetailAST ast) {
      */
     private static boolean hasUnbracedStatements(DetailAST ast) {
         final DetailAST nextSibling = ast.getNextSibling();
-        return nextSibling != null
+        final DetailAST parent = ast.getParent();
+        boolean result = false;
+
+        if (parent.getType() == TokenTypes.SWITCH_RULE) {
+            final DetailAST lastSibling = getLastSibling(ast);
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:305]: `getLastSibling` == `parent.getLastChild()`, no reason for new method here.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 1 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:548]: I've changed the method's name to  `isInSwitchRule()`, and I've used it in three more places.  Thanks for the recommendation.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 3]
```diff
@@ -476,26 +492,115 @@ private static DetailAST getLastLambdaToken(DetailAST lambda) {
...
+     *
+     * @param lambda {@link TokenTypes#LAMBDA}.
+     * @return true if current lambda belongs to a switch rule.
+     */
+    private static boolean isSwitchRuleLambda(DetailAST lambda) {
+        return lambda.getParent().getType() == TokenTypes.SWITCH_RULE;
+    }
+
+    /**
+     * Checks if current expression is a switch labeled expression . If so,
```
[the remark this answers is not in the document]
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:510]: Fixed.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 2 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: Done.

**Pull request merged**